src/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -158,6 +197,200 @@ int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onIpv4Found,
     return 0;
 }
 
+int32_t SystemNative_GetNetworkInterfaces(int32_t * interfaceCount, NetworkInterfaceInfo **interfaceList, int32_t * addressCount, IpAddressInfo **addressList )
+{
+    struct ifaddrs* head;   // Pointer to block allocated by getifaddrs().
+    struct ifaddrs* ifaddrsEntry;
+    IpAddressInfo *ai;
+    int count = 0;       // Count of entries returned by getifaddrs().
+    int ip4count = 0;    // Total number of IPv4 addresses.
+    int ip6count = 0;    // Total number of IPv6 addresses.
+    int ifcount = 0;     // Total number of unique network interface.
+    int index;
+    int socketfd = -1;
+
+    NetworkInterfaceInfo *nii;
+
+    if (getifaddrs(&head) == -1)
+    {
+        return -1;
+    }
+
+    ifaddrsEntry = head;
+    while (ifaddrsEntry != NULL)
+    {
+        count ++;
+        if (ifaddrsEntry->ifa_addr != NULL && ifaddrsEntry->ifa_addr->sa_family == AF_INET)
+        {
+            ip4count++;
+        }
+        else if (ifaddrsEntry->ifa_addr != NULL && ifaddrsEntry->ifa_addr->sa_family == AF_INET6)
+        {
+            ip6count++;
+        }
+
+        ifaddrsEntry = ifaddrsEntry->ifa_next;
+    }
+
+    // Allocate estimated space. It can be little bit more than we need.
+    void * memoryBlock = calloc((size_t)count, sizeof(NetworkInterfaceInfo));
```

Comment:
Where does this eventually get freed?  If I'm following correctly, you're allocating with calloc here and then freeing with Marshal.FreeHGlobal later?  I'm not 100% sure that's valid.  @jkotas?

Reply:
`calloc` is same underlying allocator as `malloc`. I do not see a problem here.